sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -97,13 +106,13 @@ private[sql] class CacheManager(sqlContext: SQLContext) extends Logging {
       logWarning("Asked to cache already cached data.")
     } else {
       cachedData +=
-        CachedData(
+        new CachedData(
           planToCache,
           InMemoryRelation(
             sqlContext.conf.useCompression,
             sqlContext.conf.columnBatchSize,
             storageLevel,
-            sqlContext.executePlan(query.logicalPlan).executedPlan,
+            sqlContext.executePlan(planToCache).executedPlan,
```

Comment:
I don't think we should change this line. For example, an existing Parquet dataset may be overwritten, and the new dataset has a completely different schema. When this does happen, the original code can catch the error by performing the analysis phase.